src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -54,4 +65,42 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     final protected String parserName() {
         return MatchAllQueryParser.NAME;
     }
-}
\ No newline at end of file
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        if (this.boost == 1.0f) {
+            return Queries.newMatchAllQuery();
+        }
+
+        MatchAllDocsQuery query = new MatchAllDocsQuery();
+        query.setBoost(boost);
+        return query;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof MatchAllQueryBuilder) {
+            MatchAllQueryBuilder other = (MatchAllQueryBuilder) obj;
+            return (Float.floatToIntBits(this.boost) == Float.floatToIntBits(other.boost()));
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return Float.floatToIntBits(this.boost);
+    }
```

Comment:
regarding `equals` and `hashCode`, in this specific case with a single float field, I tend to like better what my IntelliJ generates. It uses `Float.compare` in `equals`, and avoids NPEs if the argument is null:

```
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MatchAllQueryBuilder that = (MatchAllQueryBuilder) o;
        return Float.compare(that.boost, boost) == 0;
    }

    @Override
    public int hashCode() {
        return (boost != +0.0f ? Float.floatToIntBits(boost) : 0);
    }
```

Thoughts?